src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -11,25 +11,302 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
+""" Implementation of FAD test cases """
+
+import hashlib
 import json
 import os
 import sas
-from  util import winnforum_testcase, makePpaAndPalRecordsConsistent,\
- configurable_testcase, writeConfig, loadConfig, compareDictWithUnorderedLists,\
- getCertificateFingerprint
 import sas_testcase
-import hashlib
-from datetime import datetime
+from reference_models.geo import vincenty
+from sas_test_harness import SasTestHarnessServer, generateCbsdRecords, \
+    generateCbsdReferenceId, generatePpaRecords
+from util import configurable_testcase, writeConfig, loadConfig, \
+    getCertificateFingerprint, getRandomLatLongInPolygon, makePpaAndPalRecordsConsistent,compareDictWithUnorderedLists
+    winnforum_testcase
+
+class FullActivityDumpTestcase(sas_testcase.SasTestCase):
+  """ This class contains all FAD related tests mentioned in SAS TS  """
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_FAD_2_default_config(self, filename):
```

Comment:
This should be lower down, right?

Reply:
yes right, sorry, I think I have some problems with git on windows.
now it is ok?